docs/access.md

```diff
@@ -245,7 +245,121 @@ is not applicable.
 Checking if a user can perform an entity operation on group content
 -------------------------------------------------------------------
 
+OG extends the entity access control system from Drupal core so checking access
+on an entity operation is as simple as this:
+
+```php
+// Check if the given user can edit the entity, which is a group content entity.
+$access_result = $group_content_entity->access('update', $user);
+```
+
+Behind the scenes, OG implements `hook_access()` and delegates the access check
+to the `OgAccess` service, so within the context of group content this is
+equivalent to calling the following:
+
+```php
+/** @var \Drupal\og\OgAccessInterface $og_access */
+$og_access = \Drupal::service('og.access');
+$access_result = $og_access->userAccessEntityOperation('update', $group_content_entity, $user);
+```
+
+There is also a faster way to get the same result, in case you know beforehand
+to which group the group content entity belongs. The following example is more
+efficient since it doesn't need to do an expensive discovery of the groups to
+which the entity belongs:
+
+```php
+// In case we know the group entity we can use the faster method:
+$group_entity = \Drupal::entityTypeManager()->getStorage('my_group_type')->load($some_id);
+
+/** @var \Drupal\og\OgAccessInterface $og_access */
+$og_access = \Drupal::service('og.access');
+$access_result = $og_access->userAccessGroupContentEntityOperation('update', $group_entity, $group_content_entity, $user);
+```
+
 
 Altering permissions
 --------------------
 
+There are many use cases where permissions should be altered under some
+circumstances to fulfill business requirements. OG offers ways for modules to
+hook into the permission system and alter the access result.
+
+Modules can implement `hook_og_user_access_alter()` to alter group level
+permissions. Here is an example that implements a use case where groups can only
+be deleted if they are unpublished. This functionality can be toggled off by
+site administrators in the site configuration, so the example also demonstrates
+how to alter the cacheability metadata to include the config setting. The access
+result is different if this option is turned on or off, so this needs to be
+included in the cache metadata.
+
+```php
+function mymodule_og_user_access_alter(array &$permissions, CacheableMetadata $cacheable_metadata, array $context): void {
```

Comment:
@pfrenssen Lovely explanation!

I wonder if there's a good reason to keep an alter hook on the group access, but provide event subscriber for the group content access? (If you've written it in one of the issues, sorry if I've missed it)

Reply:
Very good point, there is an issue to convert it from a hook to an event: #686 but in this PR I wanted to document just the current situation. It might take some time before we get round to implementing that issue.

Reply:
:+1: